src/sealock.c

```diff
@@ -200,7 +201,27 @@ static int sealock_apply_phase_wise_result_correction(sealock_state_t *lock, tim
     lock->results.discharge_from_lake = total_volume_from_lake / new_phase_len;
     lock->results.discharge_from_sea = total_volume_from_sea / new_phase_len;
   } else {
-    // TODO: Implement handling of last 'interval', see UNST-8169.
+    // We should only get here if we ran out of rows in the timeline.
```

Comment:
I think, in the current implementation, the correction is computed one time-step too late. The correction should be computed at the last dimr time just before the phase_end. The corrected zsf results will be valid for over the period between the current dimr time until the next one; afterwards, the correction is equal zero. See slide 6 in p:\dflowfm\maintenance\JIRA\07000-07999\07576\01_stakeholder_communication\zsf_correction\20240709_discharge_correction.pptx